### PR TITLE
adam8bit for Sdxl bitandbytes 0.38.1 cuda118 windows

### PR DESCRIPTION
**Pull request by sdbds**
#447 
Old pr

### CODE REVIEW

**Comment by sdbds** [comment]
image
it works well,just slow...

**Comment by sdbds** [comment]
> Does this let you train without loss going to nan? Related Issue: #523

i had met Nan，and use --no-half-vae

**Comment by sdbds** [comment]
> 

yes,and i am testing 0.39.1。I am contacting jlllll to fix the bug in 0.39.1, which can optimize the VRAM usage of SDXL.

**Comment by sdbds** [comment]
> Thank you for this! It looks good!
> 
> Just to confirm, is the DLL from jllllll's repository?

0.38.1 it works well for long time i used.
and 0.39.1 support for bf16 training,i will test it soon.

**Comment by FurkanGozukara** [comment]
I like lion over adam
what are your parameters for training?

currently i am doing a training with Using rank adaptation algo: lokr

i used this preset 
image

image


and how can i use the generated safetensors file? with diffusers can i use?

@sdbds 

I would like to test your pr can you give me some more info please

**Comment by sdbds** [comment]
> I like lion over adam what are your parameters for training?
> 
> currently i am doing a training with Using rank adaptation algo: lokr
> 
> i used this preset image
> 
> image
> 
> and how can i use the generated safetensors file? with diffusers can i use?
> 
> @sdbds
> 
> I would like to test your pr can you give me some more info please

i recommend #631 for testing
it can use full_bf16 and PagedLion8bit
(I test most save 75% memory on ,it very useful to SDXL with bigger batch size set)

i use these↓
--cache_text_encoder_outputs
--gradient_checkpointing
--full_bf16

**Comment by FurkanGozukara** [comment]
> > I like lion over adam what are your parameters for training?
> > currently i am doing a training with Using rank adaptation algo: lokr
> > i used this preset image
> > image
> > and how can i use the generated safetensors file? with diffusers can i use?
> > @sdbds
> > I would like to test your pr can you give me some more info please
> 
> i recommend #631 for testing it can use full_bf16 and PagedLion8bit (I test most save 75% memory on PageAdamW8bit,it very useful to SDXL with bigger batch size set)
> 
> i use these↓ --cache_text_encoder_outputs --gradient_checkpointing --full_bf16

Aweosme

So you test checkpoints with comfy ui? Can you share json please?

**Comment by FurkanGozukara** [comment]
Also what learning rates and network ranks do you use for lion sdxl training

@sdbds

**Comment by sdbds** [comment]
> > > I like lion over adam what are your parameters for training?
> > > currently i am doing a training with Using rank adaptation algo: lokr
> > > i used this preset image
> > > image
> > > and how can i use the generated safetensors file? with diffusers can i use?
> > > @sdbds
> > > I would like to test your pr can you give me some more info please
> > 
> > 
> > i recommend #631 for testing it can use full_bf16 and PagedLion8bit (I test most save 75% memory on PageAdamW8bit,it very useful to SDXL with bigger batch size set)
> > i use these↓ --cache_text_encoder_outputs --gradient_checkpointing --full_bf16
> 
> Aweosme
> 
> So you test checkpoints with comfy ui? Can you share json please?

https://civitai.com/models/104515/sdxlor30costumesrevue-starlight-saijoclaudine-lora
you can download any sample image form comfyui to load json workflow

**Comment by sdbds** [comment]
> Also what learning rates and network ranks do you use for lion sdxl training
> 
> @sdbds

sorry i was busy these days so have not test with lion

**Comment by Panchovix** [comment]
@kohya-ss I tried with this PR and sadly, it seems to suffer the same issue as it is mentioned on this issue: https://github.com/kohya-ss/sd-scripts/issues/523 (On Paged AdamW 8bit)

Also, I've tested on pure AdamW 8bit (bitsandbytes==0.35.0) with cu116 on SDXL and it worked, mentioned here https://github.com/kohya-ss/sd-scripts/issues/686

Hope that helps a little!

**Comment by 2kpr** [comment]
> i recommend #631 for testing it can use full_bf16 and PagedLion8bit (I test most save 75% memory on PageAdamW8bit,it very useful to SDXL with bigger batch size set)
> 
> i use these↓ --cache_text_encoder_outputs --gradient_checkpointing --full_bf16

How did you achieve 75% memory reduction?

I am just testing the PagedAdamW8bit right now and it's basically using the same VRAM memory when training dreambooth and lora as regular AdamW/AdamW8bit or AdaFactor.

If you can tell me more about your setup or if you could show your script that would be very helpful as I am not seeing any difference between PagedAdamW8bit and AdamW/AdamW8bit.

Thanks!

**Comment by FurkanGozukara** [comment]
what is the used default weight decay for lion optimizer atm?

I am doing heavy testing with it for SDXL dreambooth training atm

i have done over 24 different experimentation so far

**Comment by sdbds** [comment]
> what is the used default weight decay for lion optimizer atm?
> 
> I am doing heavy testing with it for SDXL dreambooth training atm
> 
> i have done over 24 different experimentation so far

for lion recommend wd=0.02

**Comment by FurkanGozukara** [comment]
> > what is the used default weight decay for lion optimizer atm?
> > I am doing heavy testing with it for SDXL dreambooth training atm
> > i have done over 24 different experimentation so far
> 
> for lion recommend wd=0.02

Thanks a lot

After 30 diff experiments not satisfactory results

Do you have suggestion for sdxl dreambooth?
